VMEncryption/main/check_util.py

```diff
@@ -90,7 +94,83 @@ def is_invalid_lvm_os(self):
                     detected = True
         return detected
 
-    def is_precheck_failure(self):
+    def check_kv_url(self, test_url, message):
+        """basic sanity check of the key vault url"""
+        expected = "https://{keyvault-name}.vault.azure.net/"
+
+        if test_url is None:
+            raise Exception(message + '\nNo URL supplied')
+
+        try:
+            parse_result = urlparse.urlparse(test_url)
+        except:
+            raise Exception(message + '\nMalformed URL: ' + test_url)
+
+        if not (parse_result.scheme.lower() == "https" and \
+                re.match('[a-zA-Z0-9\-]+\.vault\.azure\.net(:443)?$', parse_result.netloc)):
+            raise Exception('\n' + message + '\nActual: ' + test_url + '\nExpected: ' + expected + "\n")
+        return
+
+    def check_kv_id(self, test_id, message):
+        """basic sanity check of the key vault id"""
+        # more strict checking would validate the full key vault id format
+        expected = "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault/vaults/{vaultname}"
+
+        if test_id is None:
+            raise Exception(message + '\nNo Resource ID supplied')
+
+        id_splits = test_id.lower().split('/')
+
+        if not (len(id_splits) >= 9 and \
+                id_splits[0] == "" and \
+                id_splits[1] == "subscriptions" and \
+                id_splits[2] != "" and \
+                id_splits[3] == "resourcegroups" and \
+                id_splits[4] != "" and \
+                id_splits[5] == "providers" and \
+                id_splits[6] == "microsoft.keyvault" and \
```

Comment:
this is a provider name, I'm not sure if lower case is valid or not, I have never seen it lower cased

Reply:
All the id_splits were converted to lower case earlier to make the checks effectively case-insensitive

Reply:
Yes, my only concern here was if the value was case sensitive and required an upper case, but the user provided it in all lower case, then this check would mark that as valid even though it wasn't.  I think we can rule that out though, as based on this list which includes casing requirements, it seems everything except for storage blob names seem to be case insensitive:  https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions